File: projects/spring-microservices/ai-service/src/main/java/com/portfolio/ai/model/AiRequest.java
package com.portfolio.ai.model;

import jakarta.validation.constraints.NotBlank;

public record AiRequest(
        @NotBlank(message = "Prompt must not be blank")
        String prompt,
        String context,
        Integer maxTokens,
        Double temperature
) {
    public AiRequest {
        if (maxTokens == null) maxTokens = 1000;
        if (temperature == null) temperature = 0.7;
    }
}

File: projects/kotlin-ai-service/src/main/kotlin/com/portfolio/ai/models/AIRequest.kt
package com.portfolio.ai.models

import kotlinx.serialization.Serializable

@Serializable
data class AIRequest(
    val prompt: String,
    val maxTokens: Int = 500,
    val temperature: Double = 0.7,
    val model: String = "gpt-4o-mini"
)
